autocleaner.py

```diff
@@ -0,0 +1,295 @@
+            templ_name, templ_ver = prci_def.get_templ_data(res.text)
+
+            if (self.box_templ_name == templ_name and
+                self.box_templ_ver == templ_ver):
+                logger.info('Box is used on %s in definition %s' %
+                            (self.branch, file_path))
+                return True
+            else:
+                return False
+
+
+    def delete_box(self):
+        del_args = ['vagrant', 'box', 'remove', self.box_templ_name,
+                    '--provider', 'libvirt', '--box-version',
+                    self.box_templ_ver]
+
+        subprocess.run(del_args, timeout=CMD_TIMEOUT)
+
+    def delete_libvirt_img(self):
+        name = self.box_templ_name.split('/')[-1]
+        del_args = ['virsh', 'vol-delete', '--pool', 'default',
+                    LIBVIRT_IMAGE_BASE.format(
+                        name=name, version=self.box_templ_ver)]
+
+        subprocess.run(del_args, timeout=CMD_TIMEOUT)
+
+
+def create_parser():
+    parser = argparse.ArgumentParser(
+        description='Perform auto-cleaning on runner')
+
+    parser.add_argument(
+        '--jobs_dir_exp', type=int, help='Number of days after which task job '
+                               'directories are deleted',
+    )
+
+    return parser
+
+
+def run(args):
+
+    vagrant_boxes = list_vagrant_boxes()
+
+    if vagrant_boxes:
+        for vagrant_box in list_vagrant_boxes():
+            logger.info('Checking if box %s is used' % vagrant_box)
+            box = Box(vagrant_box)
+            if not box.is_box_used:
+                logger.info('Deleting %s' % vagrant_box)
+                box.delete_box()
+                box.delete_libvirt_img()
+
+    if args.jobs_dir_exp:
+        res = delete_old_job_dirs(args.jobs_dir_exp)
+        if not res:
+            logger.info('No job dir qualified for erase')
+        else:
+            logger.info('Following directories deleted: ')
+            logger.info(res)
+
+    res = del_dangling_libvirt_images()
+    if not res:
+        logger.info('No dangling libvirt images found...')
+    else:
+        logger.info('Following dangling libvirt images deleted: ')
+        logger.info(res)
+
+
+def main():
+    parser = create_parser()
+    args = parser.parse_args()
+
+    try:
+        while True:
+            if not is_qemu_running():
+                logger.info('PRCI not active, stopping systemd service...')
+                stop_prci()
+                logger.info('PRCI service stopped...')
+                logger.info('Started auto-cleaning process...')
+                cleaning_process = Process(target=run, args=args)
+                cleaning_process.start()
+                cleaning_process.join(timeout=120)
+                break
+            else:
+                logger.info('PRCI active, checking again in 5 minutes...')
+                time.sleep(350)
+                continue
+    finally:
+        # lets ensure PRCI is always started when we quit
+        start_prci()
```

Comment:
this is starting the service but it is not checking that is running, what if start fails?